MeshLib/MeshGenerators/MeshLayerMapper.cpp

```diff
@@ -60,6 +61,10 @@ MeshLib::Mesh* MeshLayerMapper::createStaticLayers(MeshLib::Mesh const& mesh, st
 	std::vector<MeshLib::Node*> new_nodes(nNodes + (nLayers * nNodes));
 	std::vector<MeshLib::Element*> new_elems;
 	new_elems.reserve(nElems * nLayers);
+	MeshLib::Properties properties;
+	boost::optional<PropertyVector<int> &> materials = 
+		properties.createNewPropertyVector<int>("MaterialIDs", MeshLib::MeshItemType::Cell);
+	materials->reserve(nElems * nLayers);
```

Comment:
There is a simple pattern: if the returned value can have two states, initialized and uninitialized, it is good programming practice to check for uninitialized state.
I do not judge the implementation of `createPropertyVector` here. The changed code must be readable and unambiguous; I'm not going to find out under which circumstances the `materials` object is initialized or not. There are two possible states; the code must take care of both possibilities.

There could be a programming error in `createPropertyVector` or in any function called there in. There could be a system error. If there is a goal to write safe code then we must take care of both possible states. Otherwise we are quickly at the point where ogs5 is now.